pandas/_libs/reduction.pyx

```diff
@@ -489,6 +489,6 @@ cdef class BlockSlider:
         Ensure that we have the original blocks, blknos, and blklocs.
         """
         mgr = self.dummy._mgr
-        mgr.blocks = self.blocks
+        mgr.blocks = tuple(self.blocks)
```

Comment:
this reaching in to internal stuff is a bit sketchy....

Reply:
yes it is.  hopefully we can optimize the non-libreduction path enough to just rip this out